File: 3965-earliest-finish-time-for-land-and-water-rides-i/earliest-finish-time-for-land-and-water-rides-i.cs
public class Solution {
    public int EarliestFinishTime(int[] landStartTime, int[] landDuration, int[] waterStartTime, int[] waterDuration) {
        int minL=3000, minW=3000, res=3000, m=landStartTime.Length, n=waterStartTime.Length;
        for(int i=0; i<m; i++){
            minL=Math.Min(minL, landStartTime[i]+landDuration[i]);
        }
        for(int i=0; i<n; i++){
            minW=Math.Min(minW, waterStartTime[i]+waterDuration[i]);
            res=Math.Min(res, Math.Max(minL, waterStartTime[i])+waterDuration[i]);
        }
        for(int i=0; i<m; i++){
            res=Math.Min(res, Math.Max(minW, landStartTime[i])+landDuration[i]);
        }
        return res;
    }
}

File: 3965-earliest-finish-time-for-land-and-water-rides-i/earliest-finish-time-for-land-and-water-rides-i.java
class Solution {
    public int earliestFinishTime(int[] landStartTime, int[] landDuration, int[] waterStartTime, int[] waterDuration) {
        int minL, minW, res;
        minL=minW=res=100000000;
        int n=waterStartTime.length, m=landStartTime.length;
        for(int i=0; i<m; i++){
            minL=Math.min(minL, landStartTime[i]+landDuration[i]);
        }
        for(int i=0; i<n; i++){
            minW=Math.min(minW, waterStartTime[i]+waterDuration[i]);
            res=Math.min(res,Math.max(minL, waterStartTime[i])+waterDuration[i]);
        }
        for(int i=0; i<m; i++){
            res=Math.min(res,Math.max(minW, landStartTime[i])+landDuration[i]);
        }
        return res;
    }
}